v0/destinations/salesforce/transform.js

```diff
@@ -2,171 +2,169 @@ const get = require("get-value");
-  }
-
-  if (!rawPayload.Company || rawPayload.Company.trim() === "") {
-    rawPayload.Company = "n/a";
-  }
-
-  // Remove keys with undefined values
-  const payload = removeUndefinedValues(rawPayload);
-
-  // Get custom params from destination config
-  let customParams = getParamsFromConfig(message, destination);
-  customParams = removeUndefinedValues(customParams);
-
-  const customKeys = Object.keys(traits);
-  customKeys.forEach(key => {
-    if (
-      !defaultTraits.some(function(k) {
-        return ~k.indexOf(key);
-      }) &&
-      !ignoreMapJson.includes(key)
-    ) {
-      const val = traits[key];
-      if (val) {
-        payload[`${key}__c`] = val;
-      }
+  // get traits from the message
+  let traits = getFieldValueFromMessage(message, "traits");
+  // adjust for firstName and lastName
+  traits = { ...traits, ...getFirstAndLastName(traits, "n/a") };
+
+  // construct the payload using the mappingJson and add extra params
+  const rawPayload = constructPayload(traits, mappingJson);
+  Object.keys(traits).forEach(key => {
+    if (ignoredTraits.indexOf(key) === -1 && traits[key]) {
+      rawPayload[`${key}__c`] = traits[key];
     }
   });
 
   const response = defaultRequestConfig();
   const header = {
     "Content-Type": "application/json",
-    Authorization: authorizationData[0]
+    Authorization: authorizationData.token
   };
-
   response.method = defaultPostRequestConfig.requestMethod;
   response.headers = header;
-  response.body.JSON = { ...customParams, ...payload };
+  response.body.JSON = removeUndefinedValues(rawPayload);
   response.endpoint = targetEndpoint;
-  response.userId = message.anonymousId;
-  response.statusCode = 200;
 
   return response;
 }
 
-// Function for handling identify events
-async function processIdentify(message, destination) {
-  // Get the authorization header if not available
-  // if (!authorizationData) {
-  const authorizationData = await getSFDCHeader(destination);
-  // }
-  // start with creation endpoint, update only if Lead does not exist
-  let targetEndpoint = `${authorizationData[1]}/services/data/v${SF_API_VERSION}/sobjects/Lead`;
+// Check for externalId field under context and look for probable Salesforce objects
+// We'll make separate requests for every Salesforce Object types present under externalIds
+//
+// Expected externalId map for Contact object:
+//
+// ------------------------
+// {
+//   "type": "Salesforce-Contact",
+//   "id": "0035g000001FaHfAAK"
+// }
+// ------------------------
+//
+// We'll use the Salesforce Object names by removing "Salesforce-" string from the type field
+//
+// Default Object type will be "Lead" for backward compatibility
+async function getSalesforceIdFromPayload(message, authorizationData) {
+  // define default map
+  const salesforceMaps = [];
+
+  // get externalId
+  const externalIds = get(message, "context.externalId");
+
+  // if externalIds are present look for type `Salesforce-`
+  if (externalIds) {
```

Comment:
check if the type is array and if each array element is an object

Reply:
Added the checking. Good catch 👍